system_tests/monitoring.py

```diff
@@ -0,0 +1,133 @@
+# Copyright 2016 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest2
+
+from gcloud import _helpers
+from gcloud.environment_vars import TESTS_PROJECT
+from gcloud import monitoring
+
+
+METRIC_TYPE = 'pubsub.googleapis.com/topic/send_message_operation_count'
+METRIC_KIND = monitoring.MetricKind.DELTA
+VALUE_TYPE = monitoring.ValueType.INT64
+
+RESOURCE_TYPE = 'pubsub_topic'
+
+
+def setUpModule():
+    _helpers.PROJECT = TESTS_PROJECT
+
+
+class TestMonitoring(unittest2.TestCase):
```

Comment:
OK, but this isn't an appropriate matter to take up in relation to this
pull request. We need to get this published quickly.

On Tue, Apr 19, 2016 at 8:48 PM, Danny Hermes <notifications@github.com>
wrote:

> In system_tests/monitoring.py
> <https://github.com/GoogleCloudPlatform/gcloud-python/pull/1691#discussion_r60336468>
> :
>
> > +from gcloud.environment_vars import TESTS_PROJECT
> > +from gcloud import monitoring
> > +
> > +
> > +METRIC_TYPE = 'pubsub.googleapis.com/topic/send_message_operation_count'
> > +METRIC_KIND = monitoring.MetricKind.DELTA
> > +VALUE_TYPE = monitoring.ValueType.INT64
> > +
> > +RESOURCE_TYPE = 'pubsub_topic'
> > +
> > +
> > +def setUpModule():
> > +    _helpers.PROJECT = TESTS_PROJECT
> > +
> > +
> > +class TestMonitoring(unittest2.TestCase):
>
> For situations like that (e.g. Pub/Sub) we have a capped wait time (90s)
> and then just fail the test if the data hasn't shown up yet. Though I'm not
> sure if that translates here. As I said below, it's your choice as to how
> comprehensive you make these system tests, just offering a tidbit on how we
> work around similar issues.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/gcloud-python/pull/1691/files/adb7e102bb158a442e473166b66d5919334cfdbd#r60336468>
>